File: src/main/java/net/kingdommc/darkages/numinoustreasury/measurement/WeightUnit.java
package net.kingdommc.darkages.numinoustreasury.measurement;

public final class WeightUnit {

    public static final WeightUnit KG = new WeightUnit("kg", 1000000);
    public static final WeightUnit G = new WeightUnit("g", 1000000000);
    public static final WeightUnit ST = new WeightUnit("st", 157473);
    public static final WeightUnit LB = new WeightUnit("lb", 2204622);
    public static final WeightUnit OZ = new WeightUnit("oz", 35273952);

    private final String name;
    private final double scaleFactor;

    private WeightUnit(String name, int scaleFactor) {
        this.name = name;
        this.scaleFactor = scaleFactor;
    }

    public String getName() {
        return name;
    }

    public double getScaleFactor() {
        return scaleFactor;
    }

    public static WeightUnit getByName(String name) {
        return switch (name.toLowerCase()) {
            case "kg" -> KG;
            case "g" -> G;
            case "st" -> ST;
            case "lb" -> LB;
            case "oz" -> OZ;
            default -> null;
        };
    }

}

File: src/main/kotlin/net/arvandor/numinoustreasury/measurement/WeightUnit.kt
package net.arvandor.numinoustreasury.measurement

class WeightUnit private constructor(val name: String, scaleFactor: Int) {
    val scaleFactor: Double = scaleFactor.toDouble()

    companion object {
        val KG: WeightUnit = WeightUnit("kg", 1000000)
        val G: WeightUnit = WeightUnit("g", 1000000000)
        val ST: WeightUnit = WeightUnit("st", 157473)
        val LB: WeightUnit = WeightUnit("lb", 2204622)
        val OZ: WeightUnit = WeightUnit("oz", 35273952)

        fun getByName(name: String): WeightUnit? {
            return when (name.lowercase()) {
                "kg" -> KG
                "g" -> G
                "st" -> ST
                "lb" -> LB
                "oz" -> OZ
                else -> null
            }
        }
    }
}
